src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import { EnvironmentVariables } from '../environmentVariables'
+export const localize = nls.loadMessageBundle()
+
+// TODO: Why are messages from bundle never displayed?
+const env = process.env as EnvironmentVariables
+if (!!env.VSCODE_NLS_CONFIG) {
+    nls.config(JSON.parse(env.VSCODE_NLS_CONFIG) as nls.Options)()
+} else {
+    nls.config()()
+}
+
+export enum OutputChannelName {
+    ToolKit = 'ToolKit',
+    Lambda = 'Lambda'
+}
+const outputChannels: {[channelName: string]: vscode.OutputChannel} = {
+    [OutputChannelName.ToolKit.toString()]: vscode.window.createOutputChannel(
+        localize('AWS.channel.aws.toolkit', 'AWS Toolkit')
+    ),
+    [OutputChannelName.Lambda.toString()]: vscode.window.createOutputChannel('AWS Lambda')
+}
+
+export const getOutputChannel = (name: OutputChannelName) => outputChannels[name.toString()]
+
+// ------------------- Placeholder for upcoming logger -------------------
+// TODO: Remove this code and replace usage with upcoming src/shared/logger
+const loggerChannel = vscode.window.createOutputChannel('AWS Toolkit Logs')
+export type ErrorOrString = string | Error | number
+
+export interface Logger {
+    debug(...message: ErrorOrString[]): void
+    info(...message: ErrorOrString[]): void
+    warn(...message: ErrorOrString[]): void
+    error(...message: ErrorOrString[]): void
+}
+
+const doLog = (...message: ErrorOrString[]): void => {
+    // Ignore level for now. Log to both comnsole and output channel
+    console.log(...message)
+    loggerChannel.show()
+    loggerChannel.appendLine(message.join(' '))
+}
+
+export const getLogger = (): Logger => {
+
+    return {
+        debug: doLog,
+        info: doLog,
+        warn: doLog,
+        error: doLog
+    }
+}
+
+// ---------------------- End placeholder for upcoming logger ----------------
+
+// ------- Experimental combined output channel & traditional logger  --------------
+const logger = getLogger() as Logger | {[key: string]: (...message: ErrorOrString[]) => void}
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    msg: string,
+    nlsKey: string,
+}
+
+// See: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+const defaultTranslation = (message: string, ...args: any[]): string => {
```

Comment:
We have to do proper open source attribution of third party code. There is a process for this involving placing the unmodified source (prior to modification, if any) in the `third-party` folder, with proper references back to original location, and markdown files documenting what is brought in. Please see the third-party folder for some examples.

TODO 
- [ ] Open source attribution and importing

Reply:
Removed 3rd party code add added TODO